pkg/controllers/kube-controller-manager.go

```diff
@@ -50,13 +79,49 @@ func KubeControllerManager(cfg *config.MicroshiftConfig) {
 	if cfg.LogDir != "" {
 		args = append(args, "--log-file="+filepath.Join(cfg.LogDir, "kube-controller-manager.log"))
 	}
-	if err := command.ParseFlags(args); err != nil {
-		logrus.Fatalf("failed to parse flags: %v", err)
+
+	// fake the kube-controller-manager cobra command to parse args into controllermanager options
+	cmd := &cobra.Command{
+		Use:          "kube-controller-manager",
+		Long:         `kube-controller-manager`,
+		SilenceUsage: true,
+		RunE:         func(cmd *cobra.Command, args []string) error { return nil },
+	}
+
+	namedFlagSets := s.kubecmOptions.Flags(kubecm.KnownControllers(), kubecm.ControllersDisabledByDefault.List())
+	verflag.AddFlags(namedFlagSets.FlagSet("global"))
+	globalflag.AddGlobalFlags(namedFlagSets.FlagSet("global"), cmd.Name())
+	for _, f := range namedFlagSets.FlagSets {
+		cmd.Flags().AddFlagSet(f)
+	}
+	if err := cmd.ParseFlags(args); err != nil {
+		logrus.Fatalf("%s failed to parse flags: %v", s.Name(), err)
 	}
-	logrus.Infof("starting kube-controller-manager %s, args: %v", cfg.HostIP, args)
+}
+
+func (s *KubeControllerManager) Run(ctx context.Context, ready chan<- struct{}, stopped chan<- struct{}) error {
+	defer close(stopped)
+
+	// run readiness check
 	go func() {
-		command.Run(command, nil)
-		logrus.Fatalf("controller-manager exited")
+		healthcheckStatus := util.RetryInsecureHttpsGet("https://127.0.0.1:10257/healthz")
+		if healthcheckStatus != 200 {
+			logrus.Fatalf("Kube-controller-manager failed to start")
+		}
+
+		logrus.Infof("%s is ready", s.Name())
+		close(ready)
 	}()
 
+	c, err := s.kubecmOptions.Config(kubecm.KnownControllers(), kubecm.ControllersDisabledByDefault.List())
```

Comment:
nit: add a TODO note here mentioning https://github.com/redhat-et/microshift/blob/279375aba7f255e4d94d50098c9c87f3fbf3497e/vendor/k8s.io/kubernetes/cmd/kube-controller-manager/app/patch.go#L16

Reply:
I added the commented fn, with a note to include if/when you have an OpenShiftContext. wdyt?